src/main/java/com/GG/T9AgeCombat/service/DiceRollingService.java

```diff
@@ -45,5 +45,27 @@ int rollWithSumTakeHighest(int quantity, int takeHighestCount) {
 
         return total;
     }
+
+    Integer getFinalWithReRolls(List<Integer> resultList, Integer successThreshold, Integer reRollLessThan, Integer reRollGreaterThan) {
+        List<Integer> initialSuccesses = resultList.stream().filter(r -> r >= successThreshold).collect(Collectors.toList());
```

Comment:
Let's do this in the IF statement below so we are only doing it when we need to.

Reply:
We always need to do this because it still applies the success threshold filter to the dice. We need to do this with or without rerolls